lib/node/nodes/dot-density.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+var dot = require('dot');
+dot.templateSettings.strip = false;
+
+var Node = require('../node');
+
+var TYPE = 'dot_density';
+var PARAMS = {
+    source: Node.PARAM.NODE(Node.GEOMETRY.POLYGON),
+    points_col_name: Node.PARAM.NULLABLE(Node.PARAM.STRING),
```

Comment:
@rochoa this sounds similar to the [buffer analysis and its not-always-optional params](https://github.com/CartoDB/cartodb/issues/7502#issuecomment-223015673), right? If so I think this isn't something that could/should be tackled separately as it's not unique for this dot density analysis. 

For the moment these kind of restrictions are implemented by a developer on a per-analysis/use-case basis anyway, so the only thing we need is to know about these restrictions/combinations when implementing the form/validation rules. But if we could define them in the reference that would 👌 ofc, but like wrote above, let's take that conversation elsewhere.